pkg/operator/ceph/cluster/osd/config.go

```diff
@@ -31,11 +31,24 @@ key = %s
 `
 )
 
-func (c *Cluster) generateKeyring(osdID string, d *apps.Deployment) error {
-	resourceName := d.GetName()
-
+func (c *Cluster) generateKeyring(osdID string, deploymentName string) (string, error) {
 	user := fmt.Sprintf("osd.%s", osdID)
 	access := []string{"osd", "allow *", "mon", "allow profile osd"}
+
+	s := keyring.GetSecretStore(c.context, c.Namespace, &c.ownerRef)
```

Comment:
You are correct. I see that I missed the `c.ownerRef` on this line. I'm so sorry for the confusion.